### PR TITLE
Add Overall Status and Uptime to DashboardHome

### PR DESCRIPTION
**Pull request by lucasra1**
- expand sended stats with sendAverageUptime(duration)
- Display 24h, 7d, 30d, 90d average uptime of all monitors
- Change Uptime.vue component to support displaying average uptime of all monitors

Issue #312

### CODE REVIEW

File: server/model/monitor.js [hunk 1]
```diff
@@ -468,14 +472,78 @@ class Monitor extends BeanModel {
         } else {
             // Handle new monitor with only one beat, because the beat's duration = 0
             let status = parseInt(await R.getCell("SELECT `status` FROM heartbeat WHERE monitor_id = ?", [ monitorID ]));
-            console.log("here???" + status);
```
Comment (lucasra1) [inline server/model/monitor.js:471]: I removed it, as it seems like something, someone has forgotten to remove

**Comment by louislam** [comment]
Thanks, I was not implemented this because I am worry about the performance. Especially for 90d, the sql sum up a lot of data. 

So I think the performance proof should be required for this pr.

**Comment by gaby** [comment]
This PR will affect the performance of our `uptime-kuma` instance. We currently run the service with 2vCPU/2GB RAM, and we monitor close to 300 services/sites/servers.

**Comment by salitcao** [comment]
I want to displaying average uptime 90 days. So how can I do that ?

**Comment by deefdragon** [comment]
A note on the performance/implementation: it _might_ be possible to use a sliding summation/data collection for the graphs. In that case, you only need to add/remove the values that are now/are no longer relevant. The first request will be slow as the data is collected, but after that, it will be much faster.

**Comment by louislam** [comment]
> Thanks, I was not implemented this because I am worry about the performance. Especially for 90d, the sql sum up a lot of data.
> 
> So I think the performance proof should be required for this pr.

Closed due to this reason and seems staled.

**Comment by gaby** [comment]
@louislam what about modifying this PR to only do 24hr, 3 day, 7 day. Having the 30-90 day option are excessive.

Other option would be to make this Optional in the UI.

**Comment by louislam** [comment]
> @louislam what about modifying this PR to only do 24hr, 3 day, 7 day. Having the 30-90 day option are excessive.
> 
> Other option would be to make this Optional in the UI.

I think the performance is still questionable, this feature should be implemented after the uptime calculation performance improved.

**Pull request closed without merging**